spec/controllers/tasks_controller_spec.rb

```diff
@@ -252,6 +278,27 @@
         expect { put_update }.to change { task.reload.license.name }.to('new_license')
       end
 
+      it 'updates the tasks labels' do
+        expect { put_update }.to change { task.reload.labels }.from([existing_label]).to([new_label])
+      end
+
+      context 'when requesting new label to be created' do
+        subject(:put_update) do
+          put :update, params: {id: task.to_param, task: attributes}
+        end
+
+        let(:attributes) { {label_names: [not_existing_label_name]} }
```

Comment:
Almost, but you don't really need to redefine the put:

```suggestion
        let(:changed_attributes) { {label_names: [not_existing_label_name]} }
```

Reply:
I would also add the following expectation, since we identified the issue of not being exactly sure what Label gets created:

```ruby
        it 'creates a label with the correct name' do
          put_update
          expect(Label.last.name).to eql not_existing_label_name
        end
```

Reply:
I committed your suggestions. Thanks for the feedback!